### ISSUE TITLE
Improve industry demand calculation

### ISSUE THREAD

**Issue by timtroendle** [calliope-project/euro-calliope#63]
I copied this from https://github.com/timtroendle/possibility-for-electricity-autarky/issues/1, opened by @brynpickering .

> Industry electricity demand is currently calculated as follows:
>
> European quantity of manufactured goods per sub-class -> plant quantity of manufactured goods based on its relative contribution to sub-class emissions -> plant electricity consumption based on its energy input per unit weight of manufactured good. This is done for the three main emitting industries: ferrous metals, chloralkali, and Aluminium. 
>
> The annual energy consumption for industry, broken down into industry subclasses and by input energy source, is [published by Eurostat](https://ec.europa.eu/eurostat/web/energy/data/energy-balances) for all European countries.
>
> I would propose to keep the relative quantity of manufactured goods to be based on annual emissions, but equate that to national statistics. Annual emissions scraped from the [EU-ETS register](https://ec.europa.eu/clima/ets/welcome.do?languageCode=en) (e.g. [this plant](https://ec.europa.eu/clima/ets/ohaDetails.do?accountID=93714&action=all&languageCode=en&returnURL=resultList.currentPageNumber%3D3%26installationName%3D%26accountHolder%3D%26search%3DSearch%26permitIdentifier%3D%26form%3Doha%26searchType%3Doha%26currentSortSettings%3D%26mainActivityType%3D-1%26installationIdentifier%3D%26account.registryCodes%3DAT%26languageCode%3Den&registryCode=AT)) can then be used to have plant contributions be specific to the model year. So the workflow would look as follows:
>
> National industry energy consumption per sub-class -> plant energy consumption based on its relative contribution to sub-class emissions. This would be done for all industry sub-classes.
>
> From a first pass of the EU-ETS data, I have >5000 sites mapped, which gives a good coverage of industry in all participating countries. Non-participating countries (Switzerland, Western Balkans) could just use the national-level data from Eurostat, although Switzerland has its registry that I'm looking into how one could access...

**Comment by timtroendle** [comment]
For that to happen, we'd need to move calculation of industry demand, and residential demand between units into this repository. Currently it's taken from the results of this study: https://github.com/timtroendle/possibility-for-electricity-autarky.

Moving the demand calculation into euro-calliope makes sense as it increases the flexibility.

**Comment by brynpickering** [comment]
This should also be updated to look beyond current industry demand and consider the possibility of a decarbonised industry in future. 

This means working out what current energy demand is per industry sub-sector, allocating it to processes, and then recalculating demand based on updates to those processes (electrification, hydrogen as a reduction agent in steel-making, methanol/naphtha as input streams to plastics production). This has been implemented in our sector-coupled sibling [here](https://github.com/calliope-project/euro-calliope-2.0/blob/main/src/construct/annual_industry_demand.py)

**Comment by timtroendle** [comment]
What you describe sound like a separate issue to me. Or do we _need_ to solve both at the same time?

**Comment by brynpickering** [comment]
Yes, sorry, although it is closely linked to the issue title. This issue should really be split in two: attributing demand to industry (which can be done based on Eurostat for electricity or by inferring a decarbonised future) and regionalising that demand based on sub-sectoral demands and EU-ETS/Eurostat regional industry data. The initial comment deals with both, and what I described earlier today is specific to the former.

**Comment by timtroendle** [comment]
Blocked by #100.

**Comment by timtroendle** [comment]
Steps to be done:
1. Use subsector annual demand from #100.
2. Use emission data per plant from EU-ETS (taken from HotMaps, to be defined how to do that precisely).
3. Use emissions as a proxy for energy demand for subsectors covered by EU-ETS.

**Comment by brynpickering** [comment]
More on processing Hotmaps emissions data [here](https://github.com/calliope-project/euro-calliope-2.0/blob/main/notebooks/update_hotmaps_industry_database.ipynb)

**Comment by brynpickering** [comment]
Processing HotMaps data implemented [here](https://github.com/calliope-project/euro-calliope/commit/61f59c0eebc1183f12a732160df0448588e1d101). Mapping between JRC-IDEES and Eurostat industry subsector names [here](https://github.com/calliope-project/euro-calliope/commit/a583f06e7ac1794d3e9197d58ad4f6c0e663b01f)
